### PR TITLE
Update osgi annotations in runner

### PR DESCRIPTION
**Pull request by plutasnyy**
I updated osgi annotations in runner

## Motivation and Context
Before, annotations was marked as deprecated.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the [code style](https://github.com/Cognifide/aet/blob/master/CONTRIBUTING.md#coding-conventions) of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

---
I hereby agree to the terms of the AET Contributor License Agreement.

### CODE REVIEW

File: core/runner/src/test/java/com/cognifide/aet/runner/MessagesManagerTest.java [hunk 1]
```diff
@@ -53,21 +57,6 @@ public void createFullQueueName_expectFullName() throws Exception {
     assertThat(fullQueueName, is("AET.test"));
   }
 
-  @Test
```
Comment (plutasnyy) [inline core/runner/src/test/java/com/cognifide/aet/runner/MessagesManagerTest.java]: In my opinion, these tests was unnecessary

Comment (plutasnyy) [inline core/runner/src/test/java/com/cognifide/aet/runner/MessagesManagerTest.java]: In my opinion, these tests were unnecessary

File: core/runner/src/main/java/com/cognifide/aet/runner/MessagesManager.java [hunk 2]
```diff
@@ -98,7 +91,7 @@ public static String createFullQueueName(String name) {
   }
 
   protected String getJmxUrl() {
```
Comment (tkaik) [inline core/runner/src/main/java/com/cognifide/aet/runner/MessagesManager.java]: Is this method used anywhere?

Comment (plutasnyy) [reply core/runner/src/main/java/com/cognifide/aet/runner/MessagesManager.java]: Nope, removed

File: core/runner/src/main/java/com/cognifide/aet/runner/RunnerConfiguration.java [hunk 3]
```diff
@@ -15,132 +15,72 @@
...
-    maxMessagesInCollectorQueue = PropertiesUtil.toInteger(
-        properties.get(PARAM_MAX_MESSAGES_IN_COLLECTOR_QUEUE),
-        DEFAULT_MAX_MESSAGES_IN_COLLECTOR_QUEUE);
-    maxConcurrentSuitesCount = PropertiesUtil.toInteger(
-        properties.get(PARAM_MAX_CONCURRENT_SUITES_PROCESSED_COUNT),
-        DEFAULT_MAX_CONCURRENT_SUITES_PROCESSED_COUNT);
-
+  public void activate(RunnerConfigurationConf config) {
+    this.config = config;
     LOGGER.info(
```
Comment (Skejven) [inline core/runner/src/main/java/com/cognifide/aet/runner/RunnerConfiguration.java]: Maybe better override `toString()` in `RunnerConfigurationConf`?

File: core/runner/src/main/java/com/cognifide/aet/runner/scheduler/CollectorJobSchedulerService.java [hunk 4]
```diff
@@ -18,37 +18,44 @@
...
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.metatype.annotations.Designate;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-@Service(CollectorJobSchedulerService.class)
-@Component(description = "Collector Job Scheduler Service", label = "Collector Job Scheduler Service")
+@Component(service = CollectorJobSchedulerService.class, name = "Collector Job Scheduler Service")
+@Designate(ocd = CollectorJobSchedulerServiceConf.class)
```
Comment (Skejven) [inline core/runner/src/main/java/com/cognifide/aet/runner/scheduler/CollectorJobSchedulerService.java]: Is this empty config necessary here?

File: core/runner/pom.xml [hunk 5]
```diff
@@ -80,6 +80,18 @@
...
+      </dependency>
+    <dependency>
+      <groupId>org.osgi</groupId>
+      <artifactId>org.osgi.service.component.annotations</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.osgi</groupId>
+      <artifactId>org.osgi.annotation</artifactId>
+    </dependency>
+    <dependency>
```
Comment (Skejven) [inline core/runner/pom.xml]: Can we now remove `org.apache.felix.scr` annotation dependencies now from Runner's `pom.xml`?

Comment (plutasnyy) [reply core/runner/pom.xml]: Unfortunatelly, there is some dependency which blocked it. We should investigate it, but now I don`t know which dependency it is :/

Comment (wiiitek) [reply core/runner/pom.xml]: I have some propositions for this pull request that allow to remove these dependencies from pom:
https://github.com/Cognifide/aet/pull/309

File: core/runner/pom.xml [hunk 6]
```diff
@@ -102,14 +110,6 @@
...
     </dependency>
-    <dependency>
-      <groupId>org.powermock</groupId>
-      <artifactId>powermock-api-mockito</artifactId>
-    </dependency>
-    <dependency>
-      <groupId>org.powermock</groupId>
-      <artifactId>powermock-module-junit4</artifactId>
-    </dependency>
   </dependencies>
```
Comment (Skejven) [inline core/runner/pom.xml]: We can probably remove also:
```
<dependency>
      <groupId>org.apache.sling</groupId>
      <artifactId>org.apache.sling.commons.osgi</artifactId>
</dependency>
```

As `org.apache.sling.commons.osgi.PropertiesUtil` is no longer used.

Comment (plutasnyy) [reply core/runner/pom.xml]: Done

**Pull request merged**